### ISSUE TITLE
Enabling run_as to specify any user in a role

### ISSUE THREAD

**Issue by m-adams** [elastic/elasticsearch#59848]
At the moment, you can specify` run_as` to allow impersonation of specific users. It would be useful if this could may to something more flexible like a role. That way, as users change, there is no requirement to continually update this setting.

**Comment by tvernum** [comment]
We discussed this & the general view is:

1. This feels more suited to a group concept than roles. We don't have groups, but it seems like the desire is to be able to link the run-as privilege to a collection of users, rather than a role. We thought about groups in the past, particularly for things like sharing visualizations in Kibana, so maybe something will happen there are some point.

2. Depending on what realms exist, and how they are set up, it might be possible to do something with realm names or user metadata. That is, the run-as privilege could specify that you can run-as all users from the "corporate-ldap" realm, or all users in the "*,ou=users,..." LDAP tree.

3. For roles, it seems like there would be as much need to exclude "roles" (e.g. not admins, not superuser) etc, as there is to permit roles. If we did something here, we would need to think through what we would/wouldn't want to support and not just add "roles".

The conclusion was _Maybe...?_
Some sort of finer grained control around run-as privileges seems reasonable, but we're not sure that "mast-have-this-role" is the best way to do that.

**Comment by m-adams** [comment]
Yes, Role was not a key component. Just the only grouping is users that
seemed available.
I think in practice it would be something like an LDAP group.
